doc/ApiDocGeneration/Generate-Api-Docs.ps1

```diff
@@ -0,0 +1,305 @@
+    "recoveryservices-backup"="Recovery Services Backup"; "recoveryservices-siterecovery"="Recovery Services Site Recovery";
+    "redis"="Redis"; "relay"="Relay";
+    "reservations"="Reservations"; "resourcegraph"="Resource Graph";
+    "resources"="Resources"; "scheduler"="Scheduler";
+    "search"="Search"; "securitycenter"="Security Center";
+    "servermanagement"="Server Management"; "servicebus"="Service Bus";
+    "servicefabric"="Service Fabric"; "signalr"="signalR";
+    "sqlmanagement"="SQL Management"; "sqlvirtualmachine"="SQL Virtual Machine";
+    "storage"="Storage"; "storagecache"="Storage Cache";
+    "storagesync"="Storage Sync"; "storsimple"="Stor Simple";
+    "storsimple8000series"="Stor Simple 8000 series"; "streamanalytics"="Stream Analytics";
+    "subscription"="Subscription";
+    "trafficmanager"="Traffic Manager"; "websites"="Websites";
+}
+
+Function Get-DocFx
+{
+    [CmdletBinding()]
+    Param ($DownloadDirectory)
+
+    try 
+    {
+        Write-Verbose "Downloading DocFx..."
+        Invoke-WebRequest -Uri "https://github.com/dotnet/docfx/releases/download/v2.43.2/docfx.zip" `
+        -OutFile "docfx.zip" | Wait-Process; Expand-Archive -Path "docfx.zip" -DestinationPath "$($DownloadDirectory)/docfx/"
+    }
+    catch 
+    {
+        Write-Verbose "Failed to Download DocFx"
+        Write-Error "$_"
+        Exit 1
+    }
+}
+
+Function Get-DocTools
+{
+    [CmdletBinding()]
+    Param (
+        $DownloadDirectory,
+        $RepoRoot
+    )
+
+    try 
+    {
+        Write-Verbose "Download and Extract mdoc to Build.BinariesDirectory/mdoc"
+        Invoke-WebRequest -Uri "https://github.com/mono/api-doc-tools/releases/download/mdoc-5.7.4.9/mdoc-5.7.4.9.zip" `
+        -OutFile "mdoc.zip" | Wait-Process; Expand-Archive -Path "mdoc.zip" -DestinationPath "$($DownloadDirectory)/mdoc/"
+
+        Write-Verbose "Restore doc/ApiDocGeneration/assets/docgen.csproj, to get ECMA2Yml and popimport"
+        dotnet restore "$($RepoRoot)/doc/ApiDocGeneration/assets/docgen.csproj" /p:BuildBinariesDirectory=$DownloadDirectory
+    }
+    catch 
+    {
+        Write-Verbose "Failed to Download Mdoc"
+        Write-Error "$_"
+        Exit 1
+    }
+
+    Get-DocFx -DownloadDirectory $DownloadDirectory
+}
+
+
+
+Function Build-Docs
+{
+    [CmdletBinding()]
+    Param (
+        $ArtifactName,
+        $ServiceDirectory,
+        $ArtifactsSafeName,
+        $ArtifactsDirectoryName,
+        $LibType,
+        $RepoRoot,
+        $BinDirectory
+    )
+
+    # Include if runing Build-Docs locally
+    # Get-DocTools -DownloadDirectory $BinDirectory -RepoRoot $RepoRoot
+
+    Write-Verbose "Create variables for identifying package location and package safe names"
+    $PackageLocation = "$($ServiceDirectory)/$($ArtifactName)"
+    Write-Verbose "Package Location $($PackageLocation)"
+    $SafeName = $ArtifactsSafeName
+
+    if ($ServiceDirectory -eq '*') {
```

Comment:
We shouldn't need to special case this here. Instead we should just set ArtifactsDirectoryName in the core\ci.yml file and use that same convention as cognitiveservices and we should simply check if ArtifactDirectoryName is set and if so use it and if not use ArtifactName instead.

Reply:
I needed a general way to identify the service name for core since it is not passed from ci.yml